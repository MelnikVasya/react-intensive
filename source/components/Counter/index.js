// Core
import React from "react";
import { number } from "prop-types";

// Instruments
import Styles from "./styles.m.css";

const Counter = ({ postCount }) => (
    <section className = { Styles.counter }>{`Posts count: ${postCount}`}</section>
);

Counter.propType = {
    postCount: number,
};

Counter.defaultProps = {
    postCount: 0,
};

export default Counter;
