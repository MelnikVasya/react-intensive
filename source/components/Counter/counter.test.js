// Core
import React from "react";

// Components
import Counter from "./";
import dom from "react-test-renderer";

const render = dom.create(<Counter postCount = { 3 } />);

test("<Counter /> should render 3", () => {
    expect(render).toMatchSnapshot();
});
