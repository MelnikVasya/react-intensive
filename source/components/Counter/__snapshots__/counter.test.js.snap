// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render 3 1`] = `
<section
  className="counter"
>
  Posts count: 3
</section>
`;
